annotation-processor/src/main/java/org/hibernate/validator/ap/classchecks/ClassCheck.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.ap.classchecks;
+
+import java.util.Collection;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ExecutableElement;
+
+import org.hibernate.validator.ap.checks.ConstraintCheckError;
+
+/**
+ * <p>
+ * Implementations represent checks, which determine whether a given class
+ * is implemented in correct way.
+ * </p>
+ * <p>
+ * Implementations should be derived from {@link AbstractClassCheck} in
+ * order to implement only those check methods applicable for the element kinds
+ * supported by the check.
+ * </p>
+ *
+ *
+ * @author Marko Bekhta
+ */
+public interface ClassCheck {
+
+	/**
+	 * Checks, whether the given method is written correctly.
```

Comment:
Sorry :)
My english and punctuation is not as good as I would like them to be :)
Thanks for the patience and all those comments on "wording" and grammar :)